code/modules/mob/living/carbon/human/emote.dm

```diff
@@ -27,6 +27,25 @@
 	key = "eyebrow"
 	message = "raises an eyebrow."
 
+/datum/emote/living/carbon/human/glasses
+	key = "glasses"
+	key_third_person = "glasses"
+	message = "pushes up their glasses."
+	emote_type = EMOTE_VISIBLE
+
+/datum/emote/living/carbon/human/glasses/can_run_emote(mob/user, status_check = TRUE, intentional)
+	var/obj/eyes_slot = user.get_item_by_slot(ITEM_SLOT_EYES)
+	if(istype(eyes_slot, /obj/item/clothing/glasses/regular) || istype(eyes_slot, /obj/item/clothing/glasses/sunglasses))
+		return ..()
+	return FALSE
+
+/datum/emote/living/carbon/human/glasses/run_emote(mob/user, params, type_override, intentional)
+	. = ..()
+	if(!ishuman(user))
+		return
+	var/image/emote_animation = image('icons/mob/species/human/emote_visuals.dmi', user, "glasses")
+	flick_overlay_global(emote_animation, GLOB.clients, 1.6 SECONDS)
```

Comment:
we should do the `ishuman()` check on `can_run_emote()` instead imo, no reason to do an early return in the performative proc when we already have a proc that determines whether or not we can run it

Reply:
Oh wait I just realized that the ishuman() check is a holdover from when I was trying to use bodypart_overlays. The glasses check seems to work without it, even for mobs that can't wear things

Reply:
yeah i was wondering that since we didn't typecast to `carbon/human` but all's well that ends well